dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -62,45 +59,46 @@
   private final ImmutableList<Path> inputClasspath;
   private final SyntheticRepository syntheticRepository;
   private final ClassLoader classLoader;
+  private final ClassLoader extensionClassLoader;
   private final ImmutableSetMultimap<Path, String> jarFileToClasses;
 
   ImmutableList<Path> getInputClasspath() {
     return inputClasspath;
   }
 
+  static SyntheticRepository createSyntheticRepository(List<Path> paths) {
+    ClassPath classPath = new LinkageCheckClassPath(paths);
+    return SyntheticRepository.getInstance(classPath);
+  }
+
   static ClassDumper create(List<Path> jarFilePaths) throws IOException {
-    // Creates classpath in the same order as inputClasspath for BCEL API
-    String pathAsString =
-        jarFilePaths.stream().map(Path::toString).collect(Collectors.joining(File.pathSeparator));
-    ClassPath classPath = new ClassPath(pathAsString);
-    SyntheticRepository syntheticRepository = SyntheticRepository.getInstance(classPath);
+    URL[] jarFileUrls = new URL[jarFilePaths.size()];
+    for (int i = 0; i < jarFilePaths.size(); ++i) {
+      jarFileUrls[i] = jarFilePaths.get(i).toUri().toURL();
+    }
```

Comment:
for-loop turned out to be simpler than lambda here. (MalformedURLException extends IOException)